scripts/upload_test_results/upload_to_github.py

```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/python3
+import glob
+import os
+import argparse
+import shutil
+from git import Actor, Repo
+import json
+import sys
+import logging
+import configparser
+import urllib.request
+from github import Github
+
+logging.basicConfig(format='%(levelname)s:%(message)s', level=logging.INFO)
+
+
+def find_ver_json(commit_name):
+    # Download versions.json and find commit name
+    ver_url = 'https://testing.zephyrproject.org/daily_tests/versions.json'
+
+    try:
+        webURL = urllib.request.urlopen(ver_url)
+        data = webURL.read()
+        ver_data = json.loads(data)
+
+        for ver_name in reversed(ver_data):
+            if ver_name == commit_name:
+                logging.info('Found existing version %s', ver_name)
+                return ver_name
+        return None
```

Comment:
Sorry, I didn't state my point clearly. I mean if the function completes the `for` loop, by default it will return None here, which is not expected. You can use the following code replace `return None`. Or you can check if variable `ver_folder_name` (line 202) is equal to `None`.
```
logging.critical("ERROR: Can't find %s in versions.json file "
                         "Script execution ended.")
        sys.exit(1)
```

Reply:
Done. I added your piece of code into function `def find_ver_json()`
Then I removed check from main function after `ver_folder_name = find_ver_json(argument.commit_name)`
`if ver_folder_name is None:` Because now it is unnecessary here.